### PR TITLE
feat(evals): llm_generate for synthesizing data

### LINKED ISSUE #1309: Rails should be optional
The rails option was supposed to be optional in the function call. This is causing problems for some use cases.

def llm_eval_binary(
    dataframe: pd.DataFrame,
    template: Union[PromptTemplate, str],
    model: BaseEvalModel,
    **rails: List[str],**
    system_instruction: Optional[str] = None,
) -> List[Optional[str]]:

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #1309 

This PR adds `llm_generate` which lets you use the LLM to synthesize data per row by applying the template. This is useful for debugging or even for synthesizing data that is needed for evaluation like a bad response or GPT answer to use as a comparison.

### CODE REVIEW

File: src/phoenix/experimental/evals/functions/binary.py [hunk 1]
```diff
@@ -42,40 +44,14 @@ def llm_eval_binary(
...
-    except KeyError as e:
-        raise RuntimeError(
-            f"Error while constructing the prompts from the template and dataframe. "
-            f"The template variable {e} is not found as a column in the dataframe."
-        )
-    except Exception as e:
-        raise RuntimeError(
-            f"Error while constructing the prompts from the template and dataframe variables: {e}."
-        )
+    prompts = map_template(dataframe, eval_template)
```
Comment (mikeldking) [inline src/phoenix/experimental/evals/functions/binary.py:54]: moved to `common`

File: src/phoenix/experimental/evals/functions/common.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import pandas as pd
+
+from ..templates import PromptTemplate
+
+
+def map_template(dataframe: pd.DataFrame, template: PromptTemplate) -> "pd.Series[str]":
+    """
+    Maps over a dataframe to construct a list of prompts from a template and a dataframe.
+    """
+    # Was considering to construct the prompts and generate answers concurrently. However,
```
Comment (mikeldking) [inline src/phoenix/experimental/evals/functions/common.py:10]: Left kiko's comments

File: src/phoenix/experimental/evals/templates/template.py [hunk 3]
```diff
@@ -81,3 +81,27 @@ def _parse_variables(self) -> List[str]:
                 variables.append(variable_name)
 
         return variables
+
+
+def normalize_template(template: Union[PromptTemplate, str]) -> "PromptTemplate":
```
Comment (mikeldking) [inline src/phoenix/experimental/evals/templates/template.py:86]: centralized validation and normalization for re-use.

Comment (axiomofjoy) [reply src/phoenix/experimental/evals/templates/template.py:86]: Looks good for now. We've talked about simplifying the signature of `llm_eval_binary` to just accept an argument of type `PromptTemplate`, in which case this function would no longer be necessary.

**Review by axiomofjoy — APPROVED** [review]
Looks good. I think we're getting a proliferation of small files, we should consolidate at some point.

@RogerHYang You might want to take a look since this is touching parts of your open PR.

File: src/phoenix/experimental/evals/templates/template.py (hunk 3 above)
Comment (mikeldking) [reply src/phoenix/experimental/evals/templates/template.py:86]: Makes sense

**Pull request merged**